arch/arm/boot/dts/qcom/qcom-msm8974-htc-m8.dts

```diff
@@ -0,0 +1,332 @@
+// SPDX-License-Identifier: GPL-2.0-only
+#include "qcom-msm8974.dtsi"
+#include "qcom-pm8841.dtsi"
+#include "qcom-pm8941.dtsi"
+#include <dt-bindings/input/input.h>
+
+/ {
+	model = "HTC One (M8)";
+	compatible = "htc,m8", "qcom,msm8974";
+	chassis-type = "handset";
+
+	aliases {
+		mmc0 = &sdhc_1;
+	};
+
+	chosen {
+	};
```

Comment:
Remove this empty node if not needed

Reply:
I had to put an empty `chosen` node, for lk2nd. It does not create it if it absent.

Usually there is a `stdout-path`, but since I don't know which UART is the debug UART, there is nothing else to put here. Should I add a comment in the empty node regarding this?

Reply:
Does something break if there's no `chosen{};`?

Reply:
Yes if there is no `chosen` node, lk2nd will fail to edit the device tree before starting the kernel (due to [this code](https://github.com/msm8916-mainline/lk2nd/blob/b02d7582ef456ba1e0297d8a2beb3e098d1d8ce9/platform/msm_shared/dev_tree.c#L1406-L1411))

Reply:
Got it. I've fixed this in SoC dtsi now (and sent upstream, don't see why this shouldn't be accepted) https://github.com/msm8974-mainline/linux/commit/44c10594ebb8f0597b1cc5182d03b3acf2c92c7a so if you rebase you can drop it then :)

Reply:
OK rebased, tested and it works.